.github/workflows/auto-deploy-site.yml

```
@@ -0,0 +1,57 @@
+
+  # Deployment job
+  deploy:
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - name: Deploy to GitHub Pages
+        id: deployment
+        uses: actions/deploy-pages@v1
```

actionlint (1.7.4)

the runner of "actions/deploy-pages@v1" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)